tests/datasource/test_datasource_serialization.py

```diff
@@ -60,39 +77,148 @@
+                        "module_name": "great_expectations.datasource",
+                        "id": "dd8fe6df-254b-4e37-9c0e-2c8205d1e988",
+                        "batch_identifiers": ["default_identifier_name"],
+                    },
+                },
+                "id": "e25279d2-a809-4d1f-814c-76e892e645bf",
+            },
+            {
+                "name": "my_datasource",
+                "class_name": "Datasource",
+                "module_name": "great_expectations.datasource",
+                "data_connectors": {
+                    "my_data_connector": {
+                        "class_name": "RuntimeDataConnector",
+                        "module_name": "great_expectations.datasource",
+                        "id_": "dd8fe6df-254b-4e37-9c0e-2c8205d1e988",
+                        "batch_identifiers": ["default_identifier_name"],
+                        "name": "my_data_connector",
+                    },
+                },
+                "id_": "e25279d2-a809-4d1f-814c-76e892e645bf",
+            },
+            id="datasource_id_and_nested_data_connector_id",
+        ),
     ],
 )
 class TestDatasourceConfigSerialization:
     def test_is_serialized(
         self,
         datasource_config: DatasourceConfig,
         expected_serialized_datasource_config: dict,
+        expected_roundtrip_config: dict,
     ):
         """Datasource Config should be serialized appropriately with/without optional params."""
-        observed_dump = datasourceConfigSchema.dump(datasource_config)
-        assert observed_dump == expected_serialized_datasource_config
-
-        loaded_data = datasourceConfigSchema.load(observed_dump)
-        observed_load = DatasourceConfig(**loaded_data)
-        assert observed_load.to_json_dict() == datasource_config.to_json_dict()
+        generic_config_serialization_assertions(
+            datasource_config,
+            datasourceConfigSchema,
+            expected_serialized_datasource_config,
+            expected_roundtrip_config,
+        )
 
     def test_dict_round_trip_serialization(
         self,
         datasource_config: DatasourceConfig,
         expected_serialized_datasource_config: dict,
+        expected_roundtrip_config: dict,
     ):
         observed_dump = datasourceConfigSchema.dump(datasource_config)
 
         round_tripped = DatasourceConfig._dict_round_trip(
             datasourceConfigSchema, observed_dump
         )
 
-        assert round_tripped == datasource_config.to_json_dict()
+        assert round_tripped == expected_roundtrip_config
 
         assert (
             round_tripped.get("id_")
             == observed_dump.get("id")
             == expected_serialized_datasource_config.get("id")
+            == expected_roundtrip_config.get("id_")
         )
+
+        if (
+            hasattr(datasource_config, "data_connectors")
+            and datasource_config.data_connectors
+        ):
```

Comment:
Could we just do `datasource_config["data_connectors"]`? Connectors a required attr right why would we have to check here?

Reply:
Some of the configs don't have data_connectors and it isn't required in the config :/  https://github.com/great-expectations/great_expectations/blob/4bb482c8beffd26d5b504cc8ccb6189cb4903d30/great_expectations/data_context/types/base.py#L889-L890